acceptance/lib/puppet/acceptance/environment_utils.rb

```diff
@@ -357,6 +357,31 @@ def assert_review(review)
         end
         assert failures.empty?, "Failed Review:\n\n#{failures.join("\n")}\n"
       end
+
+      # generate a random string of 6 letters and numbers.  NOT secure
+      def random_string
+        [*('a'..'z'),*('0'..'9')].shuffle[0,6].join
+      end
+
+      # if the first test to call this has changed the environmentpath, this will cause trouble
+      #   maybe not the best idea to memoize this?
+      def environmentpath
+        @@memoized_environmentpath ||= master.puppet['environmentpath']
+      end
+
+      # create a tmpdir to hold a temporary environment bound by puppet environment naming restrictions
+      # symbolically link environment into environmentpath
+      def mk_tmp_environment(environment)
+        tmp_environment = environment.downcase + '_' + random_string
```

Comment:
yeah.  i've been mildly worried about this.  we use this method in a few places.  
we could use SecureRandom.hex with more places.  i don't think SecureRandom.uuid will be ok in many places we need a random name, and they take up too much room :-)